### PR TITLE
Gtorre collins delete api endpoint

### PR DESCRIPTION
**Pull request by gtorre**
@maddalab @roymarantz @byxorna @Primer42 

Moving forward we can just use the status endpoint to change states. Now the delete endpoint actually deletes assets, if you have the correct permissions. This change was tested.

### CODE REVIEW

**Comment by maddalab** [comment]
Build failure
```[info] Compiling 42 Scala sources to /home/travis/build/tumblr/collins/target/scala-2.10/test-classes...
[error] /home/travis/build/tumblr/collins/test/collins/controllers/AssetApiSpec.scala:193: not enough arguments for method deleteAsset: (tag: String, realDelete: Boolean)collins.controllers.actions.asset.DeleteAction.
[error] Unspecified value parameter realDelete.
[error]         Extract.from(api.deleteAsset(assetTag).apply(req)) must haveJsonData.which { txt =>
[error]                                     ^
[error] one error found
[error] (test:compile) Compilation failed
[error] Total time: 220 s, completed Jun 5, 2015 9:29:26 PM
```

**Comment by defect** [comment]
Maybe want to set the default value for realDelete to `false` to maintain compatibility and make the tests work?

**Comment by byxorna** [comment]
+1 to what @defect said. Or, change the parameter `realDelete` to be something like `purge` or `nuke`, as realDelete is kind of confusing (arent we already hitting the delete action?).

This also needs the ruby collins_client updated to handle this parameter, and a separate PR for the gh-pages to amend documentation here: https://tumblr.github.io/collins/api.html#api-asset-delete-section

**Comment by Primer42** [comment]
@gtorre - cut me a ticket explaining the doc change you want to make, so I don't forget about it.  I'm doing a big doc overhaul, and if you try to change it it'll just cause headaches.

**Comment by maddalab** [comment]
@gtorre can you make the gem change too please/

File: conf/routes [hunk 1]
```diff
@@ -75,7 +75,7 @@ DELETE  /api/asset/:tag/attribute/:attribute   collins.app.Api.deleteAssetAttrib
 POST    /api/asset/:tag/status      collins.app.Api.updateAssetStatus(tag: String)
 PUT     /api/asset/:tag             collins.app.Api.createAsset(tag: String)
 POST    /api/asset/:tag             collins.app.Api.updateAsset(tag: String)
-DELETE  /api/asset/:tag             collins.app.Api.deleteAsset(tag: String)
+DELETE  /api/asset/:tag             collins.app.Api.deleteAsset(tag: String, nuke: Boolean)
```
Comment (maddalab) [inline conf/routes]: should there be a :nuke specifier in the url?

File: support/ruby/collins-client/lib/collins/api/asset.rb [hunk 2]
```diff
@@ -21,11 +21,12 @@ def create! asset_or_tag, options = {}
...
       asset = get_asset_or_tag asset_or_tag
       parameters = {
-        :reason => get_option(:reason, options, nil)
+        :reason => get_option(:reason, options, nil),
+        :nuke   => get_option(:nuke, options, false),
       }
       parameters = select_non_empty_parameters parameters
       logger.debug("Deleting asset #{asset.tag} with parameters #{parameters.inspect}")
-      http_delete("/api/asset/#{asset.tag}", parameters, asset.location) do |response|
+      http_delete("/api/asset/#{asset.tag}?nuke=#{parameters[:nuke]}", parameters, asset.location) do |response|
```
Comment (maddalab) [inline support/ruby/collins-client/lib/collins/api/asset.rb]: I'm confused, nuke is part of the url parameter and the parameter hash?

File: conf/routes (hunk 1 above)
Comment (byxorna) [inline conf/routes]: If it is in the url, it should be optional. Excluding it from the route will mean it should be read from the body, right?

**Comment by maddalab** [comment]
@gtorre closing this PR since it no longer is the latest changes.

**Pull request closed without merging**